parsec/core/mountpoint/manager.py

```diff
@@ -268,15 +268,19 @@ async def safe_unmount_all(self):
 
 
 async def cleanup_macos_mountpoint_folder(base_mountpoint_path):
-    # In case of a crash on macOS, workspaces don't unmount correctly and leave empty directories
-    # in the default mount folder. This function is used to clean these anytime a login occurs.
-    for dirs in os.listdir(base_mountpoint_path):
-        dir_path = str(base_mountpoint_path) + "/" + str(dirs)
-        stats = os.statvfs(dir_path)
+    # In case of a crash on macOS, workspaces don't unmount correctly and leave
+    # empty mountpoints or directories in the default mount folder. This
+    # function is used to unmount or delete these anytime a login occurs.
+    for mountpoint_name in os.listdir(base_mountpoint_path):
+        mountpoint_path = str(base_mountpoint_path) + "/" + str(mountpoint_name)
+        stats = os.statvfs(mountpoint_path)
         if stats.f_blocks == 0 and stats.f_ffree == 0 and stats.f_bavail == 0:
-            await trio.run_process(["diskutil", "unmount", "force", dir_path])
-            if dirs in os.listdir(base_mountpoint_path):  # checking if there is something to delete
-                await trio.run_process(["rm", "-d", dir_path])  # otherwise an empty dir remains
+            try:
+                await trio.run_process(["diskutil", "unmount", "force", mountpoint_path])
+                if Path(mountpoint_path).is_dir():
+                    await trio.run_process(["rm", "-d", mountpoint_path])
+            except (CalledProcessError, FileNotFoundError) as exc:
+                logger.warning(f"MacOS cleanup exception", exc_info=exc)
```

Comment:
would be good to add `mountpoint_path` as part of the log (structlog allow to pass arbitrary kwargs for this)

Reply:
And what about `"Error during mountpoint cleanup: disktutil unmount failed"` for the log wording:
- no need to talk about MacOS (given we always know the platform the application runs on)
- we mention the "diskutil" command given it's a bit exotic stuff